src/Execute/DeviceViewProvider.ts

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as cp from 'child_process';
+import * as vscode from 'vscode';
+
+import {globalExecutorArray} from '../Backend/API';
+import {Command} from '../Backend/Command';
+import {DeviceSpec, supportedSpecs} from '../Backend/Spec';
+import {Balloon} from '../Utils/Balloon';
+import {Logger} from '../Utils/Logger';
+
+import {Device} from './Device';
+import {DeviceManager} from './DeviceManager';
+
+type DeviceTreeView = DeviceViewNode|undefined|void;
+
+interface DevicesManagerMap {
+  [key: string]: DeviceManager;
+}
+
+// TODO: Make this as a config.
+// This variable will save host type
+export const deviceManagerList = ['local'];
+
+export enum NodeType {
+  /**
+   * A Execution system that contains multiple Devices.
+   */
+  deviceManager,
+  /**
+   * A single device dedicated to certain device.
+   */
+  device,
+  /**
+   * A Executor that run on certain device.
+   */
+  executor,
+  /**
+   * Node for empty item.
+   */
+  none,
+}
+
+export class DeviceViewNode extends vscode.TreeItem {
+  type: NodeType;
+  managerName: string;
+  constructor(
+      public readonly labal: string,
```

Comment:
```suggestion
      public readonly label: string,
```

There are more auto-completed `labal`s. :wink:

Reply:
I will fix it. thanks!